### ISSUE TITLE
[BUG] ClassCastException: class com.salesforce.graph.vertex.VariableExpressionVertex$Single cannot be cast to class 

### ISSUE THREAD

**Issue by efwalter** [forcedotcom/sfdx-scanner#1009]
**Describe the bug**
Graph Engine identified your source and sink, but you must manually verify that you have a sanitizer in this path. Then, add an engine directive to skip the path. Next, create a Github issue for the Code Analyzer team that includes the error and stack trace. After we fix this issue, check the Code Analyzer release notes for more info. Error and stacktrace: ClassCastException: class com.salesforce.graph.vertex.VariableExpressionVertex$Single cannot be cast to class com.salesforce.graph.vertex.SoqlExpressionVertex (com.salesforce.graph.vertex.VariableExpressionVertex$Single and com.salesforce.graph.vertex.SoqlExpressionVertex are in unnamed module of loader 'app'): com.salesforce.graph.symbols.apex.system.SObjectAccessDecision.buildSanitizedValue(SObjectAccessDecision.java:171);com.salesforce.graph.symbols.apex.system.SObjectAccessDecision.executeMethod(SObjectAccessDecision.java:112);com.salesforce.graph.symbols.PathScopeVisitor.afterMethodCall(PathScopeVisitor.java:639);com.salesforce.graph.symbols.DefaultSymbolProviderVertexVisitor.afterMethodCall(DefaultSymbolProviderVertexVisitor.java:294);com.salesforce.graph.ops.expander.ApexPathExpander.visit(ApexPathExpander.java:648);com.salesforce.graph.ops.expander.ApexPathExpander.visit(ApexPathExpander.java:544)

```
 SObjectAccessDecision sobjDecision = Security.stripInaccessible( AccessType.READABLE, Database.query( query ) );
 Map<String,Account> mapCategoryAccount = new Map<String,Account>( (List<Account>) sobjDecision.getRecords() );
```
**To Reproduce**
On the second line the scanner is throwing the error

**Desktop (please complete the following information):**
 - Mac OS Ventura 13.1
 - @salesforce/sfdx-scanner 3.10.0

**Comment by rmohan20** [comment]
@efwalter Thanks for reporting this issue. We'll look into this.
As a workaround, can you try modifying your code to use an intermediary variable like this? This should unblock you for now:

```
SObjectAccessDecision sobjDecision = Security.stripInaccessible( AccessType.READABLE, Database.query( query ) );
List<Account> categoryAcctList = sobjDecision.getRecords();
Map<String,Account> mapCategoryAccount = new Map<String,Account>( categoryAcctList );
```

**Comment by efwalter** [comment]
the error still persist @rmohan20

**Comment by stephen-carter-at-sf** [comment]
Marking this as a duplicate of https://github.com/forcedotcom/sfdx-scanner/issues/1497
